lib/dao.js

```diff
@@ -167,9 +168,9 @@ DataAccessObject.create = function (data, options, cb) {
     }
   }
 
-  cb = cb || noCallback;
   data = data || {};
   options = options || {};
+  cb = cb || (Array.isArray(data) ? noCallback : utils.createPromiseCallback());
```

Comment:
In the `Array.isArray(data)` case, can we return a dummy promise object that will report a meaningful error when used via `.then` or `.catch`? The code is already in `utils.createPromiseCallback`, it just needs to be extracted into a standalone method.

```js
function createThrowingPromise(message) {
  var promise = {};
  Object.defineProperty(promise, 'then', { get: function() { throw new Error(message); });
  Object.defineProperty(promise, 'catch', { get: function() { throw new Error(message); });
  return promise;
};

// and then in dao.create()
if (!cb) {
  if (Array.isArray(data)) {
    cb = noCallback;
    cb.promise = utils.createThrowingPromise(
      'The array variant of Model.create() does not support promise-based API.');
  } else {
    cb = utils.createPromiseCallback();
  }
}
```